### PR TITLE
Remove dependency on PydanticSemanticManifest on validations

### LINKED ISSUE #587: [Feature] Remove Pydantic Implementation Dependency in Validations
### Is this your first time submitting a feature request?

- [X] I have read the [expectations for open source contributors](https://docs.getdbt.com/docs/contributing/oss-expectations)
- [X] I have searched the existing issues, and I could not find an existing issue for this feature
- [X] I am requesting a straightforward extension of existing metricflow functionality, rather than a Big Idea better suited to a discussion

### Describe the feature

The data warehouse validations module has a dependency on `PydanticSemanticManifest`, a Pydantic implementation of the `SemanticManifest`:

https://github.com/dbt-labs/metricflow/blob/f57cd53f595a84cf6a709a0bc0b5b4591b40219f/metricflow/model/data_warehouse_model_validator.py#L542

This should be updated to use `SemanticManifest` instead. There are some places in the validation code that mutate the manifest objects, so that behavior would need to be adjusted for this to be possible.

### Describe alternatives you've considered

_No response_

### Who will this benefit?

_No response_

### Are you interested in contributing this feature?

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by WilliamDee**
resolves #587

### Description
Refactored the semanticmodel validation to use hardcode SQL query instead of creating a dummy dimension and rendering SQL to query that. This removes the need to mutate the `SemanticManifest` which allows us to remove the pydantic object. Similarly, remove the pydantic dependency for the remaining objects in the Lookup classes.

<!--- 
  Before requesting review, please make sure you have:
  1. read [the contributing guide](https://github.com/dbt-labs/metricflow/blob/main/CONTRIBUTING.md),
  2. signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
  3. run `changie new` to [create a changelog entry](https://github.com/dbt-labs/metricflow/blob/main/CONTRIBUTING.md#adding-a-changelog-entry)

### CODE REVIEW

File: metricflow/model/data_warehouse_model_validator.py [hunk 1]
```diff
@@ -539,7 +538,7 @@ def run_tasks(
         return SemanticManifestValidationResults.from_issues_sequence(issues)
 
     def validate_semantic_models(
-        self, model: PydanticSemanticModel, timeout: Optional[int] = None
+        self, model: PydanticSemanticManifest, timeout: Optional[int] = None
```
Comment (tlento) [inline metricflow/model/data_warehouse_model_validator.py]: Huh. Should we need to rename the variables from model to manifest here?

Also, the typechecking failure here is disconcerting.

File: metricflow/engine/metricflow_engine.py [hunk 2]
```diff
@@ -509,14 +510,15 @@ def simple_dimensions_for_metrics(self, metric_names: List[str]) -> List[Dimensi
             linkable_dimensions_tuple,
         ) in path_key_to_linkable_dimensions.items():
             for linkable_dimension in linkable_dimensions_tuple:
-                semantic_model = self._semantic_manifest_lookup.semantic_model_lookup.get_by_reference(
+                semantic_model = semantic_model = self._semantic_manifest_lookup.semantic_model_lookup.get_by_reference(
                     linkable_dimension.semantic_model_origin
                 )
                 assert semantic_model
                 dimensions.append(
                     Dimension.from_pydantic(
```
Comment (tlento) [inline metricflow/engine/metricflow_engine.py:518]: Should we be renaming these `from_pydantic` methods to `from_semantic_model_<object>` or something?

**Review by tlento — APPROVED** [review]
Nice, thanks!

File: metricflow/engine/metricflow_engine.py (hunk 2 above)
Comment (WilliamDee) [reply metricflow/engine/metricflow_engine.py:518]: lets punt on that, I'm wondering if we could somehow clean these up and just pass the object directly?

**Pull request merged**